src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs

```diff
@@ -102,36 +69,29 @@ public bool IsCompletedOrThrow()
         private PipeCompletionCallbacks? GetCallbacks()
         {
             Debug.Assert(IsCompleted);
-            if (_callbackCount == 0)
+
+            var callbacks = _callbacks;
+            if (callbacks == null)
             {
                 return null;
             }
 
-            var callbacks = new PipeCompletionCallbacks(s_completionCallbackPool,
-                _callbackCount,
-                _exceptionInfo?.SourceException,
-                _firstCallback,
-                _callbacks);
-
-            _firstCallback = default;
             _callbacks = null;
-            _callbackCount = 0;
-            return callbacks;
+
+            return new PipeCompletionCallbacks(callbacks, _state as ExceptionDispatchInfo);
         }
 
         public void Reset()
         {
             Debug.Assert(IsCompleted);
             Debug.Assert(_callbacks == null);
-            _isCompleted = false;
-            _exceptionInfo = null;
+            _state = null;
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private void ThrowLatchedException()
+        private void ThrowLatchedException(ExceptionDispatchInfo edi)
```

Comment:
Does this method still need to exist, rather than the call site just doing edi.Throw()?

Reply:
I was wondering the same. I feel like it should be removed now